tests/cli/lib/root_account.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+check_root_account() {
+# Try to SSH to a remote machine first using root account using password-based
+# auth (this is expected to fail) and then using key-based auth with the
+# supplied username to check content of /etc/shadow and audit.log.
+#
+# use: check_root_account <user> <machine> [ssh options]
+
+    local ssh_opts="-o StrictHostKeyChecking=no"
+    local user="$1"
+    local machine="$2"
+    ssh_opts="$ssh_opts $3"
+    if [[ "$user" == "" || "$machine" == "" ]]; then
+        rlLogError "check_root_account: Missing user or machine parameter."
+        return 1
+    fi
+
+    # ssh returns 255 in case of any ssh error, so it's better to grep the specific error message
+    rlRun -t -c "ssh $ssh_opts -o PubkeyAuthentication=no root@${machine} 2>&1 | grep -i 'permission denied ('" \
+        0 "Can't SSH to '$machine' as root using password-based auth"
+    rlRun -t -c "ssh $ssh_opts ${user}@${machine} \"sudo grep -E '^root:(\*LOCK\*|!):' /etc/shadow\"" \
+        0 "root account is disabled in /etc/shadow"
+    rlRun -t -c "ssh $ssh_opts ${user}@${machine} \"sudo grep 'USER_LOGIN.*acct=\\\"root\\\".*terminal=ssh.*res=failed' /var/log/audit/audit.log\"" \
+        0 "sshd log contains entry about unsuccessful root login"
```

Comment:
Add `cat /etc/redhat-release` at the end, just because this is our historical "Check marker". In the future others may follow.